Sources/Load.php

```diff
@@ -682,120 +794,73 @@ function loadUserSettings()
-		{
-			$tfa_data = $smcFunc['json_decode']($_COOKIE[$cookiename . '_tfa'], true);
-
-			list (,, $exp) = array_pad((array) $tfa_data, 3, 0);
-
-			if (time() > $exp)
-			{
-				$_COOKIE[$cookiename . '_tfa'] = '';
-				setTFACookie(-3600, 0, '');
-			}
-		}
+}
 
-		// Create a login token if it doesn't exist yet.
-		if (!isset($_SESSION['token']['post-login']))
-			createToken('login');
-		else
-			list ($context['login_token_var'],,, $context['login_token']) = $_SESSION['token']['post-login'];
+/**
+ * Finds out if this is a spider or not.
+ *
+ * @param int $id_member
+ *
+ * @return bool|int
+ */
+function loadRobotInfo($id_member)
+{
+	global $modSettings, $sourcedir;
 
+	$possibly_robot = false;
+	if ($id_member == 0)
+	{
 		// Do we perhaps think this is a search robot? Check every five minutes just in case...
 		if ((!empty($modSettings['spider_mode']) || !empty($modSettings['spider_group'])) && (!isset($_SESSION['robot_check']) || $_SESSION['robot_check'] < time() - 300))
 		{
 			require_once($sourcedir . '/ManageSearchEngines.php');
-			$user_info['possibly_robot'] = SpiderCheck();
+			$possibly_robot = SpiderCheck();
 		}
 		elseif (!empty($modSettings['spider_mode']))
-			$user_info['possibly_robot'] = isset($_SESSION['id_robot']) ? $_SESSION['id_robot'] : 0;
+			$possibly_robot = isset($_SESSION['id_robot']) ? $_SESSION['id_robot'] : 0;
 		// If we haven't turned on proper spider hunts then have a guess!
 		else
 		{
 			$ci_user_agent = strtolower($_SERVER['HTTP_USER_AGENT']);
-			$user_info['possibly_robot'] = (strpos($_SERVER['HTTP_USER_AGENT'], 'Mozilla') === false && strpos($_SERVER['HTTP_USER_AGENT'], 'Opera') === false) || strpos($ci_user_agent, 'googlebot') !== false || strpos($ci_user_agent, 'slurp') !== false || strpos($ci_user_agent, 'crawl') !== false || strpos($ci_user_agent, 'bingbot') !== false || strpos($ci_user_agent, 'bingpreview') !== false || strpos($ci_user_agent, 'adidxbot') !== false || strpos($ci_user_agent, 'msnbot') !== false;
+			$possibly_robot = (strpos($_SERVER['HTTP_USER_AGENT'], 'Mozilla') === false && strpos($_SERVER['HTTP_USER_AGENT'], 'Opera') === false) || strpos($ci_user_agent, 'googlebot') !== false || strpos($ci_user_agent, 'slurp') !== false || strpos($ci_user_agent, 'crawl') !== false || strpos($ci_user_agent, 'bingbot') !== false || strpos($ci_user_agent, 'bingpreview') !== false || strpos($ci_user_agent, 'adidxbot') !== false || strpos($ci_user_agent, 'msnbot') !== false;
 		}
-
-		// We don't know the offset...
-		$user_info['time_offset'] = 0;
 	}
 
-	// Login Cookie times. Format: time => txt
-	$context['login_cookie_times'] = array(
-		3153600 => 'always_logged_in',
-		60 => 'one_hour',
-		1440 => 'one_day',
-		10080 => 'one_week',
-		43200 => 'one_month',
-	);
+	return $possibly_robot;
+}
+
+/**
+ * Returns an array suitable for populating `$user_info` with.
+ *
+ * @param int $id_member
+ * @param array $user_settings
+ *
+ * @return array
```

Comment:
funny note: we always define an array as return,
but you don't know what format of this array is XD